src/lib/SnackBarProvider/recoil.ts

```diff
@@ -0,0 +1,58 @@
+import { atom, useRecoilValue, useSetRecoilState } from 'recoil';
+import { useCallback } from 'react';
+import { SnackBarState } from './types';
+
+const SnackBarAtom = atom<SnackBarState>({
```

Comment:
I would suggest name this `snackBarAppState` in order to be consistent with other Atom instances